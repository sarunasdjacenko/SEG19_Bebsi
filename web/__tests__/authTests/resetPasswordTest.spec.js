import { shallowMount } from "@vue/test-utils";
import Vue from "vue";
import VueRouter from "vue-router";
// @ts-ignore
import Component from "../../src/components/auth/ResetPassword.vue";
Vue.use(VueRouter);

describe("Component", () => {
  const wrapper = shallowMount(Component);
  test("is a Vue instance", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  }),
    //snapshot test
    test("renders correctly", () => {
      expect(wrapper.element).toMatchSnapshot();
    }),
    test("has the correct buttons rendered", () => {
      const goBackBtn = wrapper.find("#backToLoginBtn");
      const resetPasswordBtn = wrapper.find("#resetPassBtn");

      expect(resetPasswordBtn).toBeDefined();
      expect(goBackBtn).toBeDefined();
    }),
    test("resets password successfully", () => {
      wrapper.setMethods({ sendResetEmail: jest.fn() });
      wrapper.find("#resetPassBtn").trigger("click");

      expect(wrapper.vm.sendResetEmail).toHaveBeenCalled();
    });
});