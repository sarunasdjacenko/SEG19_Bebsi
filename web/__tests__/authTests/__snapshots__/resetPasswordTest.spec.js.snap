// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<div
  id="resetPassword"
>
  <div
    class="container"
    style="padding-top: 10em;"
  >
    <div
      class="row"
    >
      <div
        class="col s12 m8 offset-m2"
      >
        <div
          class="login card-panel white blue-text center"
        >
          <h3>
            Reset Password
          </h3>
           
          <div
            class="input-field"
          >
            <i
              class="material-icons prefix"
            >
              email
            </i>
             
            <input
              id="email"
              name="email"
              type="text"
            />
             
            <label
              class="blue-text"
              for="email"
            >
              Email
            </label>
          </div>
           
          <router-link-stub
            event="click"
            id="backToLoginBtn"
            style="margin-right: 50px;"
            tag="a"
            to="/login"
          >
            Back to Login
          </router-link-stub>
           
          <button
            class="btn btn-large blue lighten-2 black-text"
            id="resetPassBtn"
          >
            Reset Password
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
